src/lib.rs

```diff
@@ -363,6 +368,11 @@ impl Token {
         self.expires_at
             .map_or(true, |x| Utc::now().timestamp() > x.timestamp())
     }
+
+    /// Check if the token is ready to re-authenticate automatically
+    pub fn is_capable_to_reauth(&self) -> bool {
```

Comment:
Perhaps `can_reauth` would be a better name?

Reply:
Yes, it's a better name. 

It reminds me that since `get_token` and `get_token_mut` functions not only simply returning the `Token`, but also re-authenticate automatically. The semantics of these functions has changed, it's better to have a more proper name to show the implicit flow. There some propositions:

+ `check_and_get_token`
+ `reauth_and_get_token`

Any other suggestion?

Reply:
Hmmm I get your point, but I think `get_token` isn't that bad of an idea. The token refreshing may not even be activated, so `get_token` still makes sense. Is there any occasion where you may want to get the token without refreshing it?

Reply:
Your point is reasonable, we could keep the name. But I think it's a little weird that the `get_token` is a `maybe_async` function and `get_token_mut` is `sync` function, it breaks the consistency.